### ISSUE TITLE
System.Security.Cryptography.HMACMD5 is not included in System.Security.Cryptography.Hashing.Algorithms

### ISSUE THREAD

**Issue by matsprea** [dotnet/coreclr#1359]
I found the [source of HMACMD5](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Security/Cryptography/HMACMD5.cs) in this repository, but this class is not included in the [nuget package System.Security.Cryptography.Hashing.Algorithms](https://www.nuget.org/packages/System.Security.Cryptography.Hashing.Algorithms/) that contains the others hasing algorithms

**Comment by ellismg** [comment]
The contracts for System.Security.Cryptography.Hashing.Algorithms never contained HMACMD5.  There are cases where we have source in mscorlib.dll for code that is not exposed as part of .NET Core.

I am not sure why HMACMD5 was not included in .NET Core.  It does seem to be a bit of an oversight given that we have MD5 anyway and HMAC versions of all the other hashing algorithms, but perhaps there was a rationale behind its exclusion.

/cc @bartonjs

**Comment by matsprea** [comment]
Thanks! It would be very handy to have HMACMD5 included since it's commonly used in SASL

**Comment by bartonjs** [comment]
@matsprea It seems I only replied in my head; sorry about the delay.

I tracked down a couple of people who would have been involved in the original contract creation; and no one could really identify a reason why HMACMD5 wasn't included; so its omission should feel sort of like a clerical error.

The change would come over in CoreFX, so I've made a new issue over there: https://github.com/dotnet/corefx/issues/2724.
